src/core/insights/command-queue.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  AnalyticsService,
+  Command,
+  InsightsConfig,
+  InsightsIdentity,
+} from "./types";
+import { InsightsService } from "./service";
+import * as logger from "./logger";
+
+// Queue handler that will collect commands before initialization
+export class InsightsCommandQueue implements AnalyticsService {
+  private isInitialized: boolean = false;
+  private queue: Command[] = [];
+  private debugMode: boolean = false;
+  private realImplementation: InsightsService | null = null;
+
+  constructor() {
+    // Create a proxy that will either queue commands or execute them directly
+    return new Proxy(this, {
+      get: (target: InsightsCommandQueue, prop: string) => {
+        // Return actual properties of the queue
+        if (
+          prop in target &&
+          typeof (target as InsightsCommandQueue)[prop] !== "function"
```

Comment:
We don't need these assertions, `target` is typed as `InsightsCommandQueue` above (I've verified no TS shouting locally when removed)

Reply:
Cleaned up, looks much better